rust_src/src/charset.rs

```diff
@@ -18,6 +22,20 @@ impl LispObject {
     }
 }
 
+/// Internal use only.
+/// Clear temporary charset mapping tables.
+/// It should be called only from temacs invoked for dumping.
+#[lisp_fn]
+pub fn clear_charset_maps() {
+    unsafe {
+        temp_charset_work = ptr::null_mut();
```

Comment:
The original code used `xfree`. Which means this code as written will leak memory. Rust cannot handle the C allocated memory. There are no lifetimes associated with it.

Reply:
It took me the longest time to learn xfree is defined in alloc.c and I could call it from here. I'll push it up as soon as the tests finish running locally.

Reply:
https://github.com/Wilfred/deadgrep <-- game changer for hunting around in the source.